src/utils/LockMap.ts

```diff
@@ -14,14 +14,12 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import {Lock} from "./Lock.js";
+import {Lock} from "./Lock";
 
 export class LockMap {
-    constructor() {
-        this._map = new Map();
-    }
+    private readonly _map: Map<unknown, Lock> = new Map();
```

Comment:
might be best to use a generic parameter for the key type rather than unknown? this way you can't pass different types for a key to `takeLock`...